File: Source/Bifrost.JavaScript.Specs/utils/for_StringMapping/when_resolving_a_string_with_wildcard_at_the_end.js
describe("when resolving a string with wildcard at the end", function () {
    var expectedResult = "this.is.a.wildcard.string";
    var mapping = Bifrost.StringMapping.create({
        format: "{something}/**/",
        mappedFormat: "{something}.**."
    });
    var result = mapping.resolve("this/is/a/wildcard/string");

    it("should expand input string to the mapped string", function () {
        expect(result).toBe(expectedResult);
    });
});

File: Source/Bifrost.Specs/Utils/for_StringMapping/when_resolving_a_string_with_wildcard_at_the_end.cs
using Machine.Specifications;

namespace Bifrost.Utils.for_StringMapping   
{
    public class when_resolving_a_string_with_wildcard_at_the_end
    {
        const string expected_result = "this.is.a.wildcard.string";

        static StringMapping mapping = new StringMapping(
                "{something}/**/",
                "{something}.**."
            );

        static string result;

        Because of = () => result = mapping.Resolve("this/is/a/wildcard/string");

        It should_expand_input_string_to_the_mapped_string = () => result.ShouldEqual(expected_result);
    }
}
